app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/TransactionAnalysisDelegate.kt

```diff
@@ -156,6 +162,21 @@ class TransactionAnalysisDelegate @Inject constructor(
                 allResources = resources
             )
 
+            is TransactionType.StakeTransaction -> {
+                val validators = getValidatorsUseCase(transactionType.involvedValidatorAddresses).getOrThrow()
+                transactionType.resolve(getProfileUseCase, resources, validators)
+            }
+
+            is TransactionType.UnstakeTransaction -> {
```

Comment:
All we are doing in both `TransactionType.UnstakeTransaction` and `TransactionType.ClaimStakeTransaction` cases is to return `PreviewType.Unstake(from = listOf(), to = listOf())` and `PreviewType.ClaimStake ` respectively without any use of validators. So getValidatorsUseCase is redundant in such a case.
Then would it not be better to simply return here  PreviewType.Unstake and PreviewType.ClaimStake as well as remove `UnstakeAnalysis` and `StakeClaimAnalysis` files ?